sql/planbuilder/ddl.go

```diff
@@ -420,13 +475,13 @@ func (b *Builder) buildCreateTableLike(inScope *scope, ct *ast.DDL) *scope {
 		Schema:    pkSchema,
 		IdxDefs:   idxDefs,
 		ChDefs:    checkDefs,
-		Collation: likeTable.Collation(),
-		Comment:   likeTable.Comment(),
+		Collation: coll,
+		Comment:   comment,
 	}
 
-	database := b.resolveDbForTable(ct.Table)
-
 	b.qFlags.Set(sql.QFlagSetDatabase)
+
+	outScope.setTableAlias(newTableName)
```

Comment:
I think this is probably incorrect strictly speaking, as you'll only have the columns in the last table in the list in this scope. Not sure when it matters though. @max-hoffman ?

Reply:
it's definitely an abuse of the model...a more reasonable pass might accumulate the table cols into a top-level output scope or return an empty scope